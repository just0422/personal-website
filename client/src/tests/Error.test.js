import React from 'react';
import {mount,shallow} from 'enzyme';

import ErrorModal from 'Error';

describe('Error test', () => {
	let error = {
		response: "",
		status: "",
		statusText: ""
	}

	it('should render correctly with no props', () => {
		const component = shallow(<ErrorModal />);
		expect(component).toMatchSnapshot();
	});

	it('should render correctly without component', () => {
		const component = shallow(<ErrorModal error={error}/>);
		expect(component).toMatchSnapshot();
	});

	it('should render correctly without error', () => {
		const component = shallow(<ErrorModal component="component"/>);
		expect(component).toMatchSnapshot();
	});

	it('should correctly handle refreshing the page', () => {
		const component = shallow(<ErrorModal component="component" error={error} />);
		const button = component.find('Button.refresh-button');
		window.location.reload = jest.fn();

		button.simulate('click');
		expect(window.location.reload).toHaveBeenCalled();
	});

	it('should correctly handle closing the modal', () => {
		const component = mount(<ErrorModal component="component" error={error} />);
		const button = component.find('button.close');

		button.simulate('click');
		expect(component.state('show')).toBe(false);
	});
});
