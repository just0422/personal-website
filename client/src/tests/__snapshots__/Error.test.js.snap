// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error test should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Error test should render correctly without component 1`] = `ShallowWrapper {}`;

exports[`Error test should render correctly without error 1`] = `ShallowWrapper {}`;
